micro-rdk/src/esp32/analog.rs

```diff
@@ -2,20 +2,19 @@
 use crate::common::analog::{AnalogError, AnalogReader};
 use crate::esp32::esp_idf_svc::hal::adc::{AdcChannelDriver, AdcDriver};
 use crate::esp32::esp_idf_svc::hal::gpio::ADCPin;
-use core::cell::RefCell;
-use std::rc::Rc;
+use std::sync::{Arc, Mutex};
 
 pub struct Esp32AnalogReader<'a, const A: u32, T: ADCPin> {
     channel: AdcChannelDriver<'a, A, T>,
-    driver: Rc<RefCell<AdcDriver<'a, T::Adc>>>,
+    driver: Arc<Mutex<AdcDriver<'a, T::Adc>>>,
```

Comment:
If the `Arc Mutex` wrapping access to `AnalogReader`s is in play, does the embedded `AdcDriver` for the `Esp32AnalogReader` need its own level of `Arc Mutex`'ing?

Reply:
yes, because the gRPC call grabs an analog reader and then calls read on it, meaning that the analog reader instance needs to implement Send